src/vivarium_public_health/metrics/mortality.py

```diff
@@ -0,0 +1,137 @@
+        self.start_time = self.clock()
+        self.age_bins = get_age_bins(builder)
+
+        columns_required = ['tracked', 'alive', 'age', 'entrance_time', 'exit_time',
+                            'cause_of_death', 'years_of_life_lost']
+        self.population_view = builder.population.get_view(columns_required)
+
+        builder.value.register_value_modifier('metrics', self.metrics)
+
+    def metrics(self, index, metrics):
+        pop = self.population_view.get(index)
+        pop.loc[pop.exit_time.isnull(), 'exit_time'] = self.clock()
+
+        pop = clean_cause_of_death(pop)
+        born_in_sim = pop[pop.entrance_time > self.start_time]
+        metrics.update(self.get_metrics(pop))
+        metrics.update(self.get_metrics(born_in_sim, among_born=True))
+
+        if self.by_year:
+            for year in range(self.start_time.year, self.clock().year + 1):
+                metrics.update(self.get_metrics(pop, year))
+                metrics.update(self.get_metrics(born_in_sim, year, among_born=True))
+
+        return metrics
+
+    def get_metrics(self, pop, year=None, among_born=False):
+        if year is not None:
+            start, end = pd.Timestamp(f'1-1-{year}'), pd.Timestamp(f'1-1-{year + 1}')
+        else:
+            start, end = pd.Timestamp(f'1-1-1900'), pd.Timestamp(f'1-1-2100')
+
+        out = {}
+
+        data = count_deaths(pop, self.age_bins, start, end)
+        data['person_time'] = count_person_time(pop, self.age_bins, start, end)
+        out.update(self.format_output(data, year, among_born))
+
+        return out
+
+    def format_output(self, data, year, among_born):
+        out = {}
+        born_flag = 'born_in_sim' if among_born else 'all_simulants'
+        year_flag = year if year is not None else 'all_years'
+        for i, row in data.iterrows():
+            age_group_name = self.age_bins.at[i, 'age_group_name'].replace(' ', '_')
+            for variable, value in row.iteritems():
+                label = f'{variable}_in_{year_flag}_among_{age_group_name}_{born_flag}'
+                out[label] = value
+        return out
+
+
+def count_person_time(pop, age_bins, start_time, end_time):
+    lived_in_span = pop[(start_time <= pop.exit_time) & (pop.entrance_time < end_time)]
+    # The right way to do this is np.maximum/np.minimum,
+    # but there's some bug in pandas that causes that to break.
+    entrance_time = lived_in_span.entrance_time
+    exit_time = lived_in_span.exit_time
+    exit_time.loc[end_time < exit_time] = end_time
+
+    years_in_span = (exit_time - entrance_time) / pd.Timedelta(days=365.25)
+    lived_in_span['age_at_start'] = np.maximum(lived_in_span.age - years_in_span, 0)
+
+    data = pd.Series(0, index=age_bins.index)
+
+    for group, age_bin in age_bins.iterrows():
+        start, end = age_bin.age_group_start, age_bin.age_group_end
+        in_group = lived_in_span[(start < lived_in_span.age) & (lived_in_span.age_at_start < end)]
```

Comment:
at some point, some in line comments might be really helpful to add. It took me a bit to figure out why this was < instead of <=